client/src/selectors/items-view-selectors.js

```diff
@@ -1 +1,16 @@
-const getItemsView = (state) => state.itemsView;
+import { createSelector } from "reselect";
+
+export const getItemsView = (state) => state.itemsView;
+
+const getSearchItems = (state) => getItemsView(state).todos;
+export const getIsLoading = (state) => getItemsView(state).isLoading;
+export const getError = (state) => getItemsView(state).error;
+
+
+export const getSearchItemsOnlySearched = createSelector(
```

Comment:
seems like you got the grasp on the redux flow - redux, actions, dispatches, selectors.
well done.